org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ProjectUtils.java

```diff
@@ -546,28 +545,13 @@ public static int getMaxProjectProblemSeverity() {
 			if (ProjectsManager.DEFAULT_PROJECT_NAME.equals(project.getName())) {
 				continue;
 			}
-			
+
 			try {
 				maxSeverity = Math.max(maxSeverity, project.findMaxProblemSeverity(null, true, IResource.DEPTH_ZERO));
-				if (ProjectUtils.isMavenProject(project)) {
-					IFile configFile = project.getFile(MavenProjectImporter.POM_FILE);
-					if (configFile.exists()) {
-						maxSeverity = Math.max(maxSeverity, configFile.findMaxProblemSeverity(null, true, IResource.DEPTH_ZERO));
-					}
-				} else if (ProjectUtils.isGradleProject(project)) {
-					List<String> gradleConfigs = Arrays.asList(
-						GradleProjectImporter.BUILD_GRADLE_DESCRIPTOR,
-						GradleProjectImporter.BUILD_GRADLE_KTS_DESCRIPTOR,
-						GradleProjectImporter.SETTINGS_GRADLE_DESCRIPTOR,
-						GradleProjectImporter.SETTINGS_GRADLE_KTS_DESCRIPTOR
-					);
-
-					for (String fileName : gradleConfigs) {
-						IFile configFile = project.getFile(fileName);
-						if (configFile.exists()) {
-							maxSeverity = Math.max(maxSeverity, configFile.findMaxProblemSeverity(null, true, IResource.DEPTH_ZERO));
-						}
-					}
+				List<IMarker> markers = ResourceUtils.findMarkers(project, IMarker.SEVERITY_ERROR, IMarker.SEVERITY_WARNING);
+				List<IMarker> buildFileMarkers = markers.stream().filter(marker -> isBuildFileMarker(marker, project)).collect(Collectors.toList());
+				for (IMarker marker : buildFileMarkers) {
+					maxSeverity = Math.max(maxSeverity, marker.getAttribute(IMarker.SEVERITY, 0));
```

Comment:
you can return early when severity is error

Reply:
Sure. I'll add then